darts/models/rnn_model.py

```diff
@@ -42,79 +45,64 @@ def __init__(self,
+            Tensor containing the features of the input sequence. The `input_length` is not fixed.
 
         Outputs
         -------
-        y of shape `(batch_size, out_len, output_size)`
-            Tensor containing the (point) prediction at the last time step of the sequence.
+        y of shape `(batch_size, input_length, output_size)`
+            Tensor containing the outputs of the RNN at every time step of the input sequence.
+            During training the whole tensor is used as output, whereas during prediction we only use y[:, -1, :].
+            However, this module always returns the whole Tensor.
         """
 
         super(_RNNModule, self).__init__()
 
         # Defining parameters
-        self.hidden_dim = hidden_dim
-        self.n_layers = num_layers
         self.target_size = target_size
-        num_layers_out_fc = [] if num_layers_out_fc is None else num_layers_out_fc
-        self.out_len = output_chunk_length
         self.name = name
 
         # Defining the RNN module
-        self.rnn = getattr(nn, name)(input_size, hidden_dim, num_layers, batch_first=True, dropout=dropout)
-
-        # The RNN module is followed by a fully connected layer, which maps the last hidden layer
-        # to the output of desired length
-        last = hidden_dim
-        feats = []
-        for feature in num_layers_out_fc + [output_chunk_length * target_size]:
-            feats.append(nn.Linear(last, feature))
-            last = feature
-        self.fc = nn.Sequential(*feats)
-
-    def forward(self, x):
-        # data is of size (batch_size, input_chunk_length, input_size)
+        self.rnn = getattr(nn, name)(input_size, hidden_dim, 1, batch_first=True, dropout=dropout)
+
+        # The RNN module needs a linear layer V that transforms hidden states into outputs, individually
+        self.V = nn.Linear(hidden_dim, target_size)
+
+    def forward(self, x, h=None):
+        # data is of size (batch_size, input_length, input_size)
         batch_size = x.size(0)
 
-        out, hidden = self.rnn(x)
+        # out is of size (batch_size, input_length, hidden_dim)
+        out, last_hidden_state = self.rnn(x) if h is None else self.rnn(x, h)
+        # TODO: confirm shape of out
 
-        """ Here, we apply the FC network only on the last output point (at the last time step)
+        """ Here, we apply the V matrix to every hidden state to produce the outputs
         """
-        if self.name == "LSTM":
-            hidden = hidden[0]
-        predictions = hidden[-1, :, :]
-        predictions = self.fc(predictions)
-        predictions = predictions.view(batch_size, self.out_len, self.target_size)
+        # TODO: for one layer out should be equal to hidden, make sure that's true
+        predictions = self.V(out)  # TODO: make sure this matrix multiplication is correct
+
+        # predictions is of size (batch_size, input_length, target_size)
+        predictions = predictions.view(batch_size, -1, self.target_size)
```

Comment:
Not always I think. During training it will be equal to `training_length`, at the first call of `forward` during predict it will be equal to `input_chunk_length`, and at every subsequent call to forward it will be equal to 1. 
Generally speaking our RNN module will produce an output with the same length as the input that was provided to it (in the time dimension). It's just that during predict, we only ever use the last of those outputs.